### PR TITLE
Update boss_lady_vashj.cpp

### LINKED ISSUE #15811: [3.3.5] Lady Vashj - Bossupdate needed
core:https://github.com/TrinityCore/TrinityCore/commit/ea9c112b30851b6f07f36baca2a6218f6f50cf32
DB:TDB59
update:2015_11_04_00_world.sql
boss_lady_vashj.cpp

BOSS spellID 38132(http://www.wowhead.com/spell=38132) Out of the script.

### PR DESCRIPTION
**Pull request by w5860363**
from SD2 closes #15811

### CODE REVIEW

**Comment by jackpoz** [comment]
Please provide a more detailed description than just "Update boss_lady_vashj.cpp"

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp [hunk 1]
```diff
@@ -340,6 +362,23 @@ class boss_lady_vashj : public CreatureScript
             // Return since we have no target
             if (!UpdateVictim())
                 return;
+            
+            // Check Paralyze Effect
+			if (Phase == 2)
```
Comment (jackpoz) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp]: note to whoever will merge this: tabs!

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp [hunk 2]
```diff
@@ -191,6 +192,7 @@ class boss_lady_vashj : public CreatureScript
         uint32 CoilfangStriderTimer;
         uint32 SummonSporebatTimer;
         uint32 SummonSporebatStaticTimer;
+        uint32 uint32 ParalyzecCheckTimer;
```
Comment (jackpoz) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp]: uint32 uint32 ??
also please make sure to initialized this field just like all other fields are initialized

File: src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp [hunk 3]
```diff
@@ -225,7 +227,27 @@ class boss_lady_vashj : public CreatureScript
...
+			Map::PlayerList const &PlayerList = map->GetPlayers();
+
+			for (Map::PlayerList::const_iterator i = PlayerList.begin(); i != PlayerList.end(); ++i)
+			{
+				Player* i_pl = i->GetSource();
+				if (i_pl)
+				{
+					bool check = (apply ? i_pl->HasAura(SPELL_PARALYZE) && !i_pl->HasItemCount(31088, 1, false) : i_pl->HasAura(SPELL_PARALYZE));
+
+					if (apply && i_pl->HasItemCount(31088, 1, false) && !i_pl->HasAura(SPELL_PARALYZE))
```
Comment (jackpoz) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp]: this line seems to do very similar checks as the line above, is it needed ?

Comment (w5860363) [inline src/server/scripts/Outland/CoilfangReservoir/SerpentShrine/boss_lady_vashj.cpp]: Still need

**Comment by jackpoz** [comment]
and now this pr cannot be reopened anymore because the branch was deleted...

**Comment by Nyeriah** [comment]
Code style makes it really hard to understand what is going on and it's even worse without a detailed description.

**Comment by w5860363** [comment]
@Nyeriah tainted core （http://www.wowhead.com/item=31088）Can't move but now Please excuse my English.

**Comment by Nyeriah** [comment]
This is a hack then, it should be somehow related to the spelltrigger used by that item http://collab.kpsn.org/display/tc/item_template#item_template-spelltrigger

**Comment by Nyeriah** [comment]
This looks like a hack to me, isn't there something that applies auras to players once they loot an item? (examples: sota bombs, isle of conquest bombs) http://www.wowhead.com/item=39213/massive-seaforium-charge

Perhaps I'm mistaken though, and it was added at later expansions

**Pull request closed without merging**